### ISSUE TITLE
support for Float Java type when invoking Scala methods

### ISSUE THREAD

**Issue by dkincaid** [sparklyr/sparklyr#2151]
I am unable to call a Scala method that takes a `Float` as a parameter. I get the error that `No matched method found for ....`

I am working on the sparknlp wrapper for Spark NLP and the NerDLApproach.setLr() method takes a `Float` as its argument. I looked through `invoke.scala` and don't completely understand it but didn't see anything referring to `Float` type. I get the same error with the `setPo()` method in that class that also takes a `Float`.

[NerDLApproach Scala docs](https://nlp.johnsnowlabs.com/api/index.html#com.johnsnowlabs.nlp.annotators.ner.dl.NerDLApproach)

I did setup the NerCrfApproach annotator without any problem. It has similar methods like `setL2()` but in that class they take `Double` types which work just fine. It seems to be the `Float` types that are a problem.

Some sample code:

```r
spark_version <- "2.4.0"
sc <- spark_connect(master = "local", version = spark_version)

obj <- invoke_new(sc, "com.johnsnowlabs.nlp.annotators.ner.dl.NerDLApproach")
invoke(obj, "setLr", 0.001)
```
and the error is:
```
Error: java.lang.Exception: No matched method found for class com.johnsnowlabs.nlp.annotators.ner.dl.NerDLApproach.setLr
	at sparklyr.Invoke.invoke(invoke.scala:136)
	at sparklyr.StreamHandler.handleMethodCall(stream.scala:123)
	at sparklyr.StreamHandler.read(stream.scala:66)
	at sparklyr.BackendHandler.channelRead0(handler.scala:51)
	at sparklyr.BackendHandler.channelRead0(handler.scala:4)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:284)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1359)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:935)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:138)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by kevinykuo** [comment]
The issue is that R has no float type, just doubles and integers, so we send doubles to doubles and integers to integers. For passing to Scala functions that take floats you'll need to provide a thin wrapper, e.g. https://github.com/rstudio/sparkxgb/blob/68d2322d63645b8e12174221c84fc4cbb1c54947/java/main.scala#L6-L7.

**Comment by yitao-li** [comment]
@dkincaid Alternatively you can also use the `jfloat()` function from https://github.com/sparklyr/sparklyr/pull/3066 (to be released as part of sparklyr 1.7). `jfloat()` will return a reference to a `java.lang.Float` object whose sole purpose is to be passed as a parameter of float type in `invoke*` family of functions.
